apps/system/js/app_window.js

```diff
@@ -223,6 +223,12 @@
     this.getTopMostWindow()._setActive(enable);
   };
 
+  AppWindow.prototype.inScope = function(scope) {
+    var url = new URL(this.url);
```

Comment:
Also, I learned that new URL() should be surrounded by a try/catch because it can throw an exception.